File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length();
        int l = 0;
        int r = 0;
        int maxi = 0;
        vector<int> hash(256, -1);

        while (r < n) {
            if (hash[s[r]] != -1) {
                l = max(hash[s[r]] + 1, l);
            }
            hash[s[r]] = r;
            int len = r - l + 1;
            maxi = max(len, maxi);
            r++;
        }
        return maxi;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
var lengthOfLongestSubstring = function(s) {
    let n = s.length;
    if (n === 0) return 0;

    let maxLength = 1;

    for (let i = 0; i < n; i++) {
        let count = 1;
        for (let j = i + 1; j < n; j++) {
            // Check if s[j] has occurred between s[i] to s[j - 1]
            let found = false;
            for (let k = i; k < j; k++) {
                if (s[k] === s[j]) {
                    found = true;
                    break;
                }
            }
            if (found) break;
            count++;
        }
        maxLength = Math.max(maxLength, count);
    }

    return maxLength;
};
